src/Surfnet/ServiceProviderDashboard/Application/Assembler/ServiceStatusAssembler.php

```diff
@@ -0,0 +1,195 @@
+            Service::CONNECTION_STATUS_SURFCONEXT_INFORMED => self::SERVICE_STATUS_INFO,
+            Service::CONNECTION_STATUS_ACTIVE => self::SERVICE_STATUS_SUCCESS,
+        ],
+    ];
+
+    /**
+     * @var ServiceStatusDto
+     */
+    private $serviceStatusDto;
+
+    /**
+     * ServiceStatusAssembler constructor.
+     * @param Service $service
+     * @param ServiceStatusService $serviceStatusService
+     * @param RouterInterface $router
+     */
+    public function __construct(Service $service, ServiceStatusService $serviceStatusService, RouterInterface $router)
+    {
+        $serviceLink = $router->generate('service_edit', ['id' => $service->getId()]);
+
+        $states = $this->getStates($service, $serviceStatusService);
+        $mappedStates = $this->mapStates($states);
+
+        $this->serviceStatusDto = new ServiceStatusDto(
+            $service->getName(),
+            $serviceLink,
+            $this->getEntities($service, $router),
+            $mappedStates
+        );
+
+    }
+
+    /**
+     * @return ServiceStatusDto
+     */
+    public function getDto()
+    {
+        return $this->serviceStatusDto;
+    }
+
+    /**
+     * @param Service $service
+     * @param ServiceStatusService $serviceStatusService
+     * @return array
+     */
+    private function getStates(Service $service, ServiceStatusService $serviceStatusService)
+    {
+        $states = [];
+        $type = $service->getServiceType();
+
+        $intakeStatus = $service->getIntakeStatus();
+        if ($intakeStatus != Service::INTAKE_STATUS_NOT_APPLICABLE) {
+            $states[self::SERVICE_STATE_INTAKE_CONDUCTED] = $intakeStatus;
+        }
+
+        $states[self::SERVICE_STATE_ENTITY_ON_TEST] = $serviceStatusService->getEntityStatus($service);
+
+        if ($type == Service::SERVICE_TYPE_NON_INSTITUTE) {
+            $states[self::SERVICE_STATE_CONTRACT_SIGNED] = $service->getContractSigned();
+        }
+
+        if ($type == Service::SERVICE_TYPE_INSTITUTE) {
+            $states[self::SERVICE_STATE_REPRESENTATIVE_APPROVED] = $service->getSurfconextRepresentativeApproved();
+        }
+
+        if ($service->isPrivacyQuestionsEnabled()) {
+            $states[self::SERVICE_STATE_PRIVACY_QUESTIONS] = $serviceStatusService->hasPrivacyQuestions($service);
+        }
+
+        $states[self::SERVICE_STATE_PRODUCTION_CONNECTION] = $service->getConnectionStatus();
+
+        return $states;
+    }
+
+    /**
+     * @param array $states
+     * @return array
+     */
+    private function mapStates($states)
+    {
+        $result = [];
+        foreach ($states as $name => $value) {
+            $result[$name] = $this->mapState($name, $value);
+        }
+        return $result;
+    }
+
+    /**
+     * @param Service $service
+     * @param RouterInterface $router
+     * @return array
+     */
+    private function getEntities(Service $service, RouterInterface $router)
+    {
+        $entityLinks = [];
```

Comment:
The entities need to be grouped on their connection status (production / test).
Idea: make an entity VO and collect them in this DTO

Reply:
Used the EntityList VO to accomplish this and grouped them on environment.

Reply:
Why not utilize the EntityList itself and stop keeping a entityLinks in the DTO? You could add a method to the entity list to retrieve prod/test entities.

Reply:
I've removed the separation and only return a list with VO's with the environment as parameter, then it's up to the frontend how to display the data.